### PR TITLE
Created statement analyser for Labels Node along with tests

### PR DESCRIPTION
**Pull request by nikhilsaldanha**
Hey!

Type: new feature

Link to issue: #160 

This pull request affects the following components: **(please check boxes)**

* [ ] Core
* [x] Analyzer
* [ ] Compiler
* [ ] Control Flow Graph
* [ ] Documentation

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](/.github/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change:
1. Created Analyzer for Labels and registered it to Stmt\Goto_ and Stmt\Label
2. Added it to PHPSA\Analyzer\Factory
3. Wrote 2 Tests: One with Labels and another without

### CODE REVIEW

**Comment by nikhilsaldanha** [comment]
@ovr @ddmler 
CI was failing because goto statements were not being handled by the pass method Label Analyzer. I have corrected this to:
```php
public function pass($stmt, Context $context)
{
    if ($stmt instanceof Stmt\Label) {
        $context->notice(
            'do_not_use_labels',
            'Do not use labels',
            $stmt
        );
        return true;

    } elseif ($stmt instanceof Stmt\Goto_) {
          $context->notice(
              'do_not_use_goto',
              'Do not use goto statements',
              $stmt
          );
          return true;
    }

    return false;
}
```

Since Label Statement Analyzer has been registered to both goto and label statements, should I be testing for goto statements as well in the DoNotUseLabels test? Maybe something like this:

```php
<?php

namespace Tests\Analyze\Fixtures\Statement;

class DoNotUseLabels {
    public function testLabel()
    {
        marker1:
        $a = 1;

        marker2:

        marker3:
        goto marker1;

    }
    public function testNoLabel()
    {
        $a = 2;

    }
}

?>
----------------------------
[
    {
        "type": "do_not_use_labels",
        "message": "Do not use labels",
        "file": "DoNotUseLabels.php",
        "line": 7
    },
    {
        "type": "do_not_use_labels",
        "message": "Do not use labels",
        "file": "DoNotUseLabels.php",
        "line": 10
    },
    {
        "type": "do_not_use_labels",
        "message": "Do not use labels",
        "file": "DoNotUseLabels.php",
        "line": 12
    },
    {
        "type": "do_not_use_goto",
        "message": "Do not use goto statements",
        "file": "DoNotUseLabels.php",
        "line": 13
    }
]
```

File: src/Analyzer/Pass/Statement/DoNotUseLabels.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+class DoNotUseLabels implements Pass\ConfigurablePassInterface, Pass\AnalyzerPassInterface
+{
+    /**
+     * @param Stmt\Label $stmt
+     * @param Context $context
+     * @return bool
+     */
+    public function pass($stmt, Context $context)
+    {
+
```
Comment (sergeyklay) [inline src/Analyzer/Pass/Statement/DoNotUseLabels.php]: Could you please remove blank line

File: tests/analyze-fixtures/Statement/DoNotUseLabels.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Tests\Analyze\Fixtures\Statement;
+
+class DoNotUseLabels {
```
Comment (sergeyklay) [inline tests/analyze-fixtures/Statement/DoNotUseLabels.php]:
```php
class DoNotUseLabels
{
```

File: src/Analyzer/Pass/Statement/DoNotUseLabels.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    /**
+     * @return TreeBuilder
+     */
+    public function getConfiguration()
+    {
+        $treeBuilder = new TreeBuilder();
+        $treeBuilder->root('do_not_use_labels')
+            ->canBeDisabled()
+        ;
```
Comment (sergeyklay) [inline src/Analyzer/Pass/Statement/DoNotUseLabels.php]:
```php
->canBeDisabled();
```

File: tests/analyze-fixtures/Statement/DoNotUseLabels.php (hunk 2 above)
Comment (nikhilsaldanha) [inline tests/analyze-fixtures/Statement/DoNotUseLabels.php]: Okay, done!

**Comment by nikhilsaldanha** [comment]
@sergeyklay I've made all the changes that you'd mentioned and the CIs are passing as well

**Comment by ddmler** [comment]
@nikhilsaldanha sorry i think we misunterstood each other. We wanted you to add this to DoNotUseGoto analyzer (this one exists already) and not make a new one. What your new analyzer does is exactly what DoNotUseGoto should do so just copy it over and we can merge.

**Comment by nikhilsaldanha** [comment]
@ddmler Sorry, I misunderstood what was required. I've made the fix

**Comment by ddmler** [comment]
Thanks @nikhilsaldanha 
Check issues page if you want to help more. :cake:

**Pull request merged**